File: CodingBat/Java/Array-1/maxEnd3.java
/**
 * Given an array of ints length 3, figure out which is larger, the first or last element in the array, and set all the other elements to be that value. Return the changed array.

maxEnd3([1, 2, 3]) → [3, 3, 3]
maxEnd3([11, 5, 9]) → [11, 11, 11]
maxEnd3([2, 11, 3]) → [3, 3, 3]

 */

public int[] maxEnd3(int[] nums) {
    int larger = Math.max(nums[0], nums[nums.length-1]);
    int changed[] = {larger, larger, larger};
    return changed;
  }
  

File: CodingBat/Python/List-1/max_end3.py
"""

Given an array of ints length 3, figure out which is larger, the first or last element in the array, and set all the other elements to be that value. Return the changed array.


max_end3([1, 2, 3]) → [3, 3, 3]
max_end3([11, 5, 9]) → [11, 11, 11]
max_end3([2, 11, 3]) → [3, 3, 3]

@author unobatbayar

"""

def max_end3(nums):
  def findLargest():
    if nums[0] > nums[2]:
      return nums[0]
    return nums[2]

  largest = findLargest()
  changed_array = [largest, largest, largest]
  return changed_array